pkg/vacuum/vacuum.go

```diff
@@ -275,24 +294,50 @@ func (e *Engine) Run(ctx context.Context) {
 		}
 		numberVacuumConnections.Set(0)
 	}()
+
+	for {
```

Comment:
It took me a while to understand what this outer `for` is supposed to achieve — prevent starvation of either workload. Am I right in this assessment? If so, I think, it would be easier to understand if there were just a single top-level loop taking batches of rows in a round-robin fashion from one source and then another.

Reply:
Yes, it is intended to prevent either workload from being completely starved. See [the comment above](https://github.com/timescale/promscale/pull/1804#discussion_r1052649470).

We could do round-robin, but that would give equal preference to both workloads. In my opinion, the `vacuumChunksToFreeze` workload is more important because it's more urgent. These chunks are older and closer to wraparound.

You could argue that if the `vacuumChunksToFreeze` starves out the other, then it might cause more chunks to fall into the `vacuumChunksToFreeze` bucket, but my hunch is that this is unlikely. But I could be wrong.

WDYT?

Reply:
I suppose you are right about `vacuumChunksToFreeze` being more important, but my concern is primarily about structure. I think it is possible to allocate more time for `vacuumChunksToFreeze` even if we round-robin between several workloads. I made an attempt at illustrating my suggestion here https://github.com/timescale/promscale/pull/1810/files

Reply:
I like that there is less duplicate code, and I think you're weighted round-robin is likely better too.
I think there are a couple of issues: it will keep trying both workloads when one is finished, and I'm not sure there is logic to exit the loop when both finish. I can fix that though. Thanks!

Reply:
okay, how's this change?